src/TeamSpeak3QueryApi/Query/QueryClient.cs

```diff
@@ -82,19 +106,52 @@ public async Task<CancellationTokenSource> Connect()
             _writer = new StreamWriter(_ns) { NewLine = "\n" };
 
             IsConnected = true;
+            OnConnected?.Invoke(this, EventArgs.Empty);
 
-            await _reader.ReadLineAsync().ConfigureAwait(false);
+
+            var headline = await _reader.ReadLineAsync().ConfigureAwait(false);
+            if (headline != "TS3")
+            {
+                throw new QueryProtocolException("Telnet Query isn't a valid Teamspeak Query");
+            }
             await _reader.ReadLineAsync().ConfigureAwait(false); // Ignore welcome message
             await _reader.ReadLineAsync().ConfigureAwait(false);
 
             return ResponseProcessingLoop();
         }
 
+        public CancellationTokenSource ConnectSsh(string username, string password)
```

Comment:
Doesn't TS support public key auth via SSH?

Reply:
First of all teamspeak support key Auth. But I am not sure if there is a public key Auth. Need investigation